src/renderer/canvas/node.ts

```diff
@@ -97,9 +110,43 @@ const drawNodeLabel = <N extends INodeBase, E extends IEdgeBase>(
   const center = node.getCenter();
   const distance = node.getBorderedRadius() * (1 + DEFAULT_LABEL_DISTANCE_SIZE_FROM_NODE);
 
+  let labelX = center.x;
+  let labelY = center.y;
+  let labelTextAlign: LabelTextAlign;
+  let labelTextBaseline: LabelTextBaseline;
+
+  switch (node.getLabelAlignment()) {
+    case NodeLabelAligment.BOTTOM:
+      labelY += distance;
+      labelTextAlign = LabelTextAlign.CENTER;
+      labelTextBaseline = LabelTextBaseline.TOP;
+      break;
+    case NodeLabelAligment.TOP:
+      labelY -= distance;
+      labelTextAlign = LabelTextAlign.CENTER;
+      labelTextBaseline = LabelTextBaseline.BOTTOM;
+      break;
+    case NodeLabelAligment.LEFT:
+      labelX -= distance;
+      labelTextAlign = LabelTextAlign.RIGHT;
+      labelTextBaseline = LabelTextBaseline.MIDDLE;
+      break;
+    case NodeLabelAligment.RIGHT:
+      labelX += distance;
+      labelTextAlign = LabelTextAlign.LEFT;
+      labelTextBaseline = LabelTextBaseline.MIDDLE;
+      break;
+    default:
+      labelY += distance;
+      labelTextAlign = LabelTextAlign.CENTER;
+      labelTextBaseline = LabelTextBaseline.TOP;
+      break;
+  }
```

Comment:
Actually, this should be part of Label logic. I would propose the following:

```
NodeLabelPosition.{TOP, BOTTOM, LEFT, RIGHT} // default = BOTTOM
NodeLabelAlignment.{LEFT, RIGHT, CENTER} // default = CENTER

EdgeLabelPosition.{CENTER} // default = CENTER
EdgeLabelAlignment.{LEFT, RIGHT, CENTER} // default = CENTER
```

And then Label logic receives position + alignment and handles the rest. Please take care of multiple lines because it is not working well. Check the image below for the "top".

![Screenshot 2023-08-16 at 22 22 31](https://github.com/memgraph/orb/assets/2602830/36d5ada1-6b4b-4535-8812-434e1788a74d)

Reply:
Node logic should just handle the start position: `x`, `y` and give that to the Label logic. Label should take the start position + label position + label alignment + text lines to figure out where each line should be.